2024FALL-SWPP/team-project-for-2024-fall-swpp-team-16
# feat: Title and Stage Selection

## Description
Added Stage selection page and also title page. Title page loads stage selection when any key is pressed. Stage selection scene allows user to choose between 'unlocked' stages

## Related Issue


## Changes Made

- [x] New Scene: `Title Scene Updated` Any key pressed --> Stage Scene is loaded
- [x] New Scene: `Stage Scene` Stage scene loads with player's current level selected. 
- [x] New Dummy Scenes in `01.Scenes/ Stages` (just to test loading)
- [x] Script: `TitleScreenManager` --> press enter to load level selection manager
- [x] Script: `LevelSelectionManager` --> Slides upward to be visible when Awoken. Arrow key used to select between unlocked stages (represented by notes, each with `LevelSelectionNoteController` script). (Only unlocked stages can be selected). Enter pressed to load new scene
- [x] Script: `LevelSelectionNoteController` --> assigns appropriate color to note images based on selected (blue), unlocked (black) and locked (grey). 
- [x] Camera effect when the stage starts / ends
- [x] Connect `LevelSelectionManager` with `StageManager`

## How to Test This Branch

StageScene scene: In inspector, Canvas / Level Selection Manager Object has Level selection manager script. Assign level notes to correct level notes. 
In inspector Canvas / Level Selection Manager / Stages , each # stage object is a stagenote that has LevelSelectionNoteController script to track stage state (unlocked, selected, stage number). Adjust locked/ unlocked state  and run scene to see that only unlocked notes can be selected between.

Lastly, In `01.Scenes/Stages`, there are dummy scenes ("Test Stage# JS" with the # as 1 or 2) that will be loaded. When stage scene selects stage 1 or 2, it should open the dummy scene. Otherwise, it should have 'scene doesnt exist' on console

<img width="634" alt="Screenshot 2024-12-03 at 2 24 04 PM" src="https://github.com/user-attachments/assets/1b851a1d-de1a-4900-af5a-676d77007646">

## Screenshots (if applicable)
![stage-selection](https://github.com/user-attachments/assets/d6c30ecf-4d4f-4005-836b-7b71e4716f02)


## Additional Notes
